### PR TITLE
Added atreides-01b

### PR DESCRIPTION
**Pull request by abcdefg30**
Depends on #8117.

TODO:
- [x] Map preview
- [x] Briefing video

### CODE REVIEW

**Comment by penev92** [comment]
As in A1V1, the wait for bounties on Hard in A1V2 is ridiculously long and annoying (feels even longer in this one).
Also the seconday wormspawner here is even worse, because there is no safe area to harvest in this map.

**Comment by abcdefg30** [comment]
Updated as #8117 was merged.

File: mods/d2k/maps/atreides-01b/atreides01b.lua [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+	harkonnen = Player.GetPlayer("Harkonnen")
+
+	color = HSLColor.New(170, 255, 200)
+
+	if Map.Difficulty == "Normal" then
+		HarkonnenAttackWaves = 5
+		ToHarvest = 3000
+	elseif Map.Difficulty == "Hard" then
+		HarkonnenAttackWaves = 12
+		ToHarvest = 3500
```
Comment (obrakmann) [inline mods/d2k/maps/atreides-01b/atreides01b.lua]: You could put these values in a hash like you do for the reinforcements as well, if you wanted to.

File: mods/d2k/maps/atreides-01b/atreides01b.lua [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	end
+
+	if player.Resources > ToHarvest - 1 then
+		player.MarkCompletedObjective(GatherSpice)
+	end
+
+	-- player has no Wind Trap
+	if (player.PowerProvided <= 20 or player.PowerState ~= "Normal") and DateTime.GameTime % DateTime.Seconds(32) == 0 then
+		HasPower = false
+		Media.DisplayMessage(Messages[2], "")
```
Comment (obrakmann) [inline mods/d2k/maps/atreides-01b/atreides01b.lua]: Don't forget to add the `Mentat` prefix for these messages as well.

**Comment by obrakmann** [comment]
Looks good to me. :+1: with the noted changes, plus the map preview.

File: mods/d2k/maps/atreides-01b/atreides01b.lua [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	-- player has no Refinery and no Silos
+	if HasPower and player.ResourceCapacity == 0 and DateTime.GameTime % DateTime.Seconds(32) == 0 then
+		Media.DisplayMessage(Messages[3], "")
+	end
+
+	if HasPower and player.Resources > player.ResourceCapacity * 0.8 and DateTime.GameTime % DateTime.Seconds(32) == 0 then
+		Media.DisplayMessage(Messages[4], "")
+	end
+
+	UserInterface.SetMissionText("Harvested resources: " .. player.Resources .. "/" .. ToHarvest, color)
```
Comment (Mailaender) [inline mods/d2k/maps/atreides-01b/atreides01b.lua]: Please include the changes from https://github.com/OpenRA/OpenRA/pull/8292.

File: mods/d2k/maps/atreides-01b/atreides01b.lua [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+end
+
+WorldLoaded = function()
+	player = Player.GetPlayer("Atreides")
+	harkonnen = Player.GetPlayer("Harkonnen")
+
+	color = HSLColor.New(170, 255, 200)
```
Comment (Mailaender) [inline mods/d2k/maps/atreides-01b/atreides01b.lua]: Please include the changes from https://github.com/OpenRA/OpenRA/pull/8292.

File: mods/d2k/maps/atreides-01b/map.yaml [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+	PlayerReference@Creeps:
+		Name: Creeps
+		NonCombatant: True
+		Race: Random
+	PlayerReference@Atreides:
+		Name: Atreides
+		Playable: true
+		Race: atreides
+		LockRace: true
+		ColorRamp: 170,255,127
```
Comment (Mailaender) [inline mods/d2k/maps/atreides-01b/map.yaml]: Please include the changes from https://github.com/OpenRA/OpenRA/pull/8292.

File: mods/d2k/maps/atreides-01b/atreides01b.lua (hunk 1 above)
Comment (abcdefg30) [inline mods/d2k/maps/atreides-01b/atreides01b.lua]: Done.

**Comment by abcdefg30** [comment]
Updated and resolved all nits (except the preview).

**Comment by penev92** [comment]
:+1: from me once you add the 2 mission previews that I sent you. (The preview for Atreides01b that I made with help from @x-a-n-a-x made the one I had done for Atreides01a look bad, so I redid that. Now they match perfectly.)

**Comment by abcdefg30** [comment]
Added the map previews.

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/0f70e2aa8a6acec0dbb9333134a31188eea06ae3...d1c4d60afde9c506cfb62565071c0fce60f5a247)

**Pull request merged**